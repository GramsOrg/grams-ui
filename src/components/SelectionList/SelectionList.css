.middle {
    vertical-align: middle !important;
}

.rtl {
    direction: rtl;
}

.rtl-content {
    padding-right: 1em !important;
}
