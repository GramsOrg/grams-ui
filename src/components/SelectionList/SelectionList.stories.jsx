import React from 'react';

import { action } from "@storybook/addon-actions";

import SelectionList from './SelectionList.tsx';

export default {
  title: 'General/SelectionList',
  component: SelectionList,
  parameters: {
    componentSubtitle: 'A configurable selection list component with icons and labels.',
    storyshots: { disable: false },
  },
  argTypes: {
    dir: {
      control: 'radio',
      options: ['ltr', 'rtl']
    },
    inverted: {
      control: 'boolean'
    }
  }
};

const Template = ({dir, inverted, ...props}) => {
  return (
    <SelectionList
      dir={dir}
      inverted={inverted}
      onSelect={action("Trigger onSelect()")}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  dir: 'ltr',
  inverted: false
}

const options = [
  {
    id: 'new',
    title: 'Create new profile',
    description: 'Start a new profile from scratch',
    icon: 'user plus',
    color: 'black'
  }, {
    id: 'import-backup',
    title: 'Import profile backup',
    description: 'Import a .grams backup file',
    icon: 'upload',
    color: 'black'
  }, {
    id: 'import-seed',
    title: 'Import profile seed',
    description: 'Import your profile using the 24-word seed',
    icon: 'pencil',
    color: 'black'
  }, {
    id: 'ledger',
    title: 'Use Ledger wallet',
    description: 'Pair Grams with your ledger wallet',
    icon: 'usb',
    color: 'black'
  }, 
]

export const CreateProfileOptions = Template.bind({});
CreateProfileOptions.args = {
  dir: 'ltr',
  inverted: false,
  options: options
}

export const RTL = Template.bind({});
RTL.args = {
  dir: 'rtl',
  inverted: false,
  options: options
}
