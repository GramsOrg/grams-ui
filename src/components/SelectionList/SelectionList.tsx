import React from 'react';

import { Icon, List, SemanticICONS, SemanticCOLORS } from 'semantic-ui-react';

import './SelectionList.css';

/**
 * Interface for the option object used in SelectionList.
 * @interface
 */
export interface SelectionOption {
  id: string;
  title: string;
  description?: string;
  icon?: SemanticICONS;
  color?: SemanticCOLORS;
}

/**
 * Props for SelectionList component.
 * @interface
 */
export interface ISelectionListProps {

  /**
   * Defines the direction of the component
   */
  dir?: 'ltr' | 'rtl';

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /** 
   * A list of options for the user to choose from.
   */
  options?: SelectionOption[];

  /**
   * A callback function to be called when an option is selected.
   */
  onSelect?: (option: SelectionOption) => void;
}

const defaultProps = {
  dir: 'ltr',
  inverted: false,
  options: [{
    id: 'test',
    title: 'Hello',
    description: 'start now',
    icon: 'world',
    color: 'blue'
  }]
};

const SelectionList = (props: ISelectionListProps) => {

  return (
    <List relaxed='very' selection className={props.dir}>
      {props?.options?.map((option) => (
        <List.Item
          key={option.id}
          onClick={() => props?.onSelect?.(option)}
        >
          <Icon 
            className='middle'
            size='big'
            name={option.icon}
            color={option.color}
          />
          <List.Content className={`${props.dir}-content`}>
            <List.Header as='h2'>{option.title}</List.Header>
            <List.Description>{option.description}</List.Description>
          </List.Content>
        </List.Item>
      ))}
    </List>
  );
};

SelectionList.defaultProps = defaultProps;

export default SelectionList;
