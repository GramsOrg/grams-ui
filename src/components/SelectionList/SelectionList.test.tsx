/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import SelectionList from './SelectionList';

describe('SelectionList Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<SelectionList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
