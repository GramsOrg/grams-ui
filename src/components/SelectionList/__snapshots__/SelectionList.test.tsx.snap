// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionList Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui selection very relaxed list ltr"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="blue world big icon middle"
      />
      <div
        class="content ltr-content"
      >
        <h2
          class="header"
        >
          Hello
        </h2>
        <div
          class="description"
        >
          start now
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
